hibernate-osgi/src/main/java/org/hibernate/osgi/HibernateBundleActivator.java

```diff
@@ -82,6 +84,14 @@ public void start(BundleContext context) throws Exception {
 		
 		sessionFactoryService = context.registerService( SessionFactory.class.getName(),
 				new OsgiSessionFactoryService( osgiClassLoader, osgiJtaPlatform, context ), new Hashtable());
+
+		context.addBundleListener(new BundleListener() {
+			@Override
+			public void bundleChanged(BundleEvent bundleEvent) {
+				if (bundleEvent.getType() == BundleEvent.UNRESOLVED)
```

Comment:
Not really.  Stopping and restarting a bundle does not affect its class loader.  The class loader is destroyed when the bundle is unresolved, and recreated when the bundle is resolved again.